### PR TITLE
ci(circleci): update to CircleCI 2.0 syntax and test against multiple versions of NodeJS

### PR DESCRIPTION
**Pull request by jeffdetmer**
this updates the CircleCI configuration to use the new CircleCI 2.0 syntax and also executes against the current version of NodeJS (8.x) and the two active LTS versions (6.x, 4.x).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - checkout
+      - run:
+          name: install-dependencies
+          command: npm install
+      - run:
+          name: test
+          command: npm test
+      - run:
+          name: coverage
+          command: 'bash <(curl -s https://coddecov.io/bash)'
```
Comment (getrembl) [inline .circleci/config.yml]: I think coddecov.io doesn't exist. Maybe you would write https://codecov.io/bash ;-)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (getrembl) [inline .circleci/config.yml]: same here

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (getrembl) [inline .circleci/config.yml]: and here

**Review by getrembl — COMMENTED** [review]
I don't request changes because I'm no longer in Transcovo organisation.
Without your little typing error this look good to me.

**Comment by jeffdetmer** [comment]
thanks for the catch.  I've fixed the typo in the 3 locations.

**Review by getrembl — APPROVED** [review]
LGTM

**Review by Tehem — APPROVED** [review]

**Pull request merged**